evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -491,6 +491,51 @@ def test_fit_correct_inputs(
     assert_series_equal(expected_y, mock_ohe_fit_transform.call_args[0][1])
 
 
+@patch("evalml.pipelines.components.Transformer.fit_transform", autospec=True)
+@patch("evalml.pipelines.components.OneHotEncoder.transform", autospec=True)
+@patch("evalml.pipelines.components.Imputer.transform", autospec=True)
+@patch("evalml.pipelines.components.LabelEncoder.transform", autospec=True)
+def test_component_graph_fit_transform(
+    mock_label_encoder_transform,
+    mock_imputer_transform,
+    mock_ohe_transform,
+    mock_fit_transform,
+    example_graph,
+    example_graph_with_transformer_last_component,
+    X_y_binary,
+):
+    X, y = X_y_binary
+    component_graph = ComponentGraph(example_graph)
+    component_graph.instantiate()
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "Cannot call fit_transform() on a component graph because the final component is an Estimator. Use fit_and_transform_all_but_final instead."
+        ),
+    ):
+        component_graph.fit_transform(X, y)
+
+    component_graph = ComponentGraph(example_graph_with_transformer_last_component)
+    component_graph.instantiate()
+    ones_df = pd.DataFrame(np.ones(pd.DataFrame(X).shape))
+
+    def fit_transform_side_effect(self, X, y):
+        self._is_fitted = True
+        return ones_df
+
+    mock_fit_transform.side_effect = fit_transform_side_effect
+    mock_label_encoder_transform.return_value = ones_df
+    mock_imputer_transform.return_value = ones_df
+    mock_ohe_transform.return_value = ones_df
+
+    component_graph.fit_transform(X, y)
+
+    assert mock_label_encoder_transform.call_count == 2
+    assert mock_imputer_transform.call_count == 1
+    assert mock_ohe_transform.call_count == 1
+    assert mock_fit_transform.call_count == 2
+
```

Comment:
Did you ever figure out why we had to mock the encoder call twice and the other transformer components once? Not blocking but just curious to see if you found out.

Reply:
haven't got a chance to dig deeper there unfortunately

Reply:
Would definitely like to know the answer here before moving on!

Reply:
Found out the `LabelEncoder` class overrides its `fit_transform` method with `self.fit().transform()`, so when the LabelEncoder transform mock is called, it actually triggers 2 calls to its `transform()` unlike the OHE and Imputer